test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -0,0 +1,342 @@
+        controllerURI = ctlURIs.get(0);
+
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+
+        controller = new ControllerImpl(ControllerImplConfig.builder()
+                .clientConfig(clientConfig)
+                .maxBackoffMillis(5000).build(), executor);
+        streamManager = StreamManager.create(clientConfig);
+
+        assertTrue("Creating scope", streamManager.createScope(SCOPE));
+        assertTrue("Creating stream", streamManager.createStream(SCOPE, STREAM,
+                StreamConfiguration.builder()
+                        .scalingPolicy(ScalingPolicy.fixed(1))
+                        .retentionPolicy(RetentionPolicy.bySizeBytes(MIN_SIZE_IN_STREAM, MAX_SIZE_IN_STREAM)).build()));
+    }
+
+    @After
+    public void tearDown() {
+        streamManager.close();
+        ExecutorServiceHelpers.shutdown(executor);
+        ExecutorServiceHelpers.shutdown(streamCutExecutor);
+    }
+
+    @Test
+    public void multipleSubscriberCBRTest() throws Exception {
+        final ClientConfig clientConfig = Utils.buildClientConfig(controllerURI);
+
+        @Cleanup
+        EventStreamClientFactory clientFactory = EventStreamClientFactory.withScope(SCOPE, clientConfig);
+        @Cleanup
+        EventStreamWriter<String> writer = clientFactory.createEventWriter(STREAM, new JavaSerializer<>(),
+                EventWriterConfig.builder().build());
+
+        // Write events.
+        log.info("Writing event e1 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e2 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e3 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e4 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e5 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e6 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        log.info("Writing event e7 to {}/{}", SCOPE, STREAM);
+        writer.writeEvent(SIZE_30_EVENT).join();
+
+        @Cleanup
+        ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(SCOPE, clientConfig);
+        readerGroupManager.createReaderGroup(READER_GROUP_1, ReaderGroupConfig.builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints()
+                .stream(Stream.of(SCOPE, STREAM)).build());
+        readerGroupManager.createReaderGroup(READER_GROUP_2, ReaderGroupConfig.builder()
+                .retentionType(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT)
+                .disableAutomaticCheckpoints()
+                .stream(Stream.of(SCOPE, STREAM)).build());
```

Comment:
NIT : we can assert here to check the status of `createReaderGroup`

Reply:
done